### PR TITLE
Iteration one

### PR DESCRIPTION
**Pull request by zliibbe**
closes issue #1 , issue #4,  issue #5, and issue #6 

## Purpose:
This PR completes iteration one of Travel Tracker. 

*What does this merge do? Check all that apply.*
- [x] New feature (non-breaking change which adds functionality)
- [ ] Improves an existing feature
- [ ] Fixes a bug
      - Bug fixed:
- [ ] Refactor
      - Code refactored:
- [ ] Documentation update

## Description: What does this PR do?
As a traveler, I am able to see a dashboard showing all of my trips, regardless of status. I can also see the total amount a user has spent on trips in the current year (2022), which is calculated including a 10% travel agent's fee. Testing is complete except for testing for trip class, which is missing several methods testing. The methods work for multiple users and were tested on user #1, user #7, and user #14, which all displayed accurately to the DOM. APICalls fetch data from the local server and all code 

## Instructions for reviewer:
*Where should the reviewer start?* Overall class structure and apiCalls are my main concerns. I feel confident in my methods as I've tested them with multiple users from the server and regardless of trip status, everything displays on the site as expected. Ready to get to work on iteration 2, truly refine and get HTML/CSS looking good, useable, and accessible
*How should this be manually tested?* 
*Any background context you want to provide?* Long work day, feeling quite empty as of this PR. Please merge once you've reviewed so I can continue working. 

##Screenshots:
image
image

## Issues:
*Are there any concerns, issues, or bugs in this branch? If so describe them.*
Testing still needs to be completed for the trip class. CSS and HTML are in need of refinement, but are sufficient and meet the specs for everything being displayed.

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have used tests to prove my fix is effective or that my feature works
- [x] My changes generate no new warnings
- [ ] New and existing unit tests pass locally with my change

## Future Steps:

Complete unit testing, iteration 2

### CODE REVIEW

File: test/traveler-test.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+            "name": "Ham Leadbeater",
+            "travelerType": "relaxer"
+            };
+        travelerTwoData = {
+            "id": 14,
+            "name": "Ellynn Kyne",
+            "travelerType": "history buff"
+            }
+        traveler1 = new Traveler(travelerOneData)
+        traveler2 = new Traveler(travelerTwoData)
```
Comment (CHaiz15) [inline test/traveler-test.js:29]: I'm confused by the need for 2 different travelers for testing here. Any reason for having 2 or can we pass all the tests with just 1?

File: test/trip-test.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            "travelerType": "relaxer"
+            };
+        travelerTwoData = {
+            "id": 2,
+            "name": "Rachael Vaughten",
+            "travelerType": "thrill-seeker"
+            };
+
+        traveler1 = new Traveler(travelerOneData)
+        traveler2 = new Traveler(travelerTwoData)
```
Comment (CHaiz15) [inline test/trip-test.js:29]: Same Q as travelers tests

File: src/domUpdates.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+//querySelectors
+const tripList = document.querySelector('.trip-list')
+const greeting = document.querySelector('.greeting')
+const annualSpending = document.querySelector('.annual-expenditure')
+
+const domUpdates = {
+    userGreeting(user) {
+        let theUser = user.getFirstName()
+        greeting.innerText = `Welcome back, ${theUser}!`
```
Comment (CHaiz15) [inline src/domUpdates.js:11]: Does this function need the whole user object passed in if it only uses the name?

File: src/domUpdates.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    },
+
+    displayAnnualSpending(user) {
+        annualSpending.innerText = `You've spent ${user.annualSpent()} in ${dayjs().format('YYYY')}`
```
Comment (CHaiz15) [inline src/domUpdates.js:15]: Same question. Does this need the whole user object if it only uses the annual spent value?

File: src/apiCalls.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+    .then(response => {
+    if (!response.ok) {
+        throw new Error('Error with requesting information.');
+      } else {
+       return response.json();
+      }
+    }).catch(error => {
+        error = 'TypeError: Failed to fetch' ? errorMsg.innerText = 'Server is not currently running' : 
+        errorMsg.innerText = 'Error encountered. See console.'
+        console.warn(error)
```
Comment (CHaiz15) [inline src/apiCalls.js:12]: I'm confused by what's happening here. You're assigning error to a string and also errorMsg to a string based of a ternary?¿

**Review by CHaiz15 — CHANGES_REQUESTED** [review]
Overall this is awesome and a super good start!! Just some various minor things I commented on. There's also a lot of commented out console logs that are something that should be deleted before submitting a PR. But again, solid start and definitely going in the right direction 👍👍👍

File: src/scripts.js [hunk 6]
```diff
@@ -1,11 +1,61 @@
...
+        if (parseInt(id) === traveler.id) {
+            return traveler
+        }
+    })
+}
+
+
 
+const updateDOM = (userData, tripsData, destinationsData) => {
+    domUpdates.userGreeting(userData)
```
Comment (Andrew-Musselman) [inline src/scripts.js:55]: I like the way you've condensed this into one function to be called after the promis.all

File: src/trip.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+    costOfTrip(destinationData) {
+        const destinationObj = this.findDestination(destinationData)
+        const flightCost = destinationObj.estimatedFlightCostPerPerson * this.travelers
+        const lodgingCost = destinationObj.estimatedLodgingCostPerDay * this.duration * this.travelers
+        // console.log(destinationObj, flightCost, lodgingCost, "destination | flightCost | lodingCost")
+        return Math.round(1.1 * (flightCost + lodgingCost))
+    }
+}
+
```
Comment (Andrew-Musselman) [inline src/trip.js:29]: I like that you have methods here to find the cost and the destinations. It does look like the find destination one is written like an es6 function and not a method

File: src/domUpdates.js (hunk 3 above)
Comment (zliibbe) [reply src/domUpdates.js:11]: I see what you're saying. But I don't know how to pass in just a dynamic string. My getFirstName() doesn't doesn't exist outside of domUpdates. Does that mean I should import it here in order for that to work? I think it does, so I'll give that a try and consider this comment a "rubber duck"

**Review by Andrew-Musselman — APPROVED** [review]
Looks Good!

**Pull request merged**